vip-helpers/vip-caching.php

```diff
@@ -670,8 +670,8 @@ function wpcom_vip_set_old_slug_redirect_cache( $link ) {
 	return $link;
 }
 function wpcom_vip_flush_wp_old_slug_redirect_cache( $post_id, $post, $post_before ) {
-	// Don't bother if slug hasn't changed.
-	if ( $post->post_name == $post_before->post_name ) {
+	// Don't bother if the slug or date hasn't changed.
+	if ( $post->post_name == $post_before->post_name && $post->post_date == $post_before->post_date ) {
```

Comment:
No, should be `&&` for sure. We only want to exit early if at least one of those has changed. Otherwise, if the name was the same it would exit early and not clear cache if the date changed.

Reply:
I see how this could be confusing though. Another way to phrase it would be:

```
if ( $post->post_name == $post_before->post_name && $post->post_date == $post_before->post_date ) {
  // The page name and post date haven't changed, just return early.
  return;
}
```